Authentication and User Profiles

PotatoPlank opened the pull request:
This is the first step towards moving towards an API structure. 
All authentication is now using an API and database migrations. User groups and profiles were added for future functionality.

SKroell committed:
Laravel test workflow for Github actions

SKroell committed:
Workflow for deploying commits to master branch.

SKroell committed:
Update ExampleTest

Update ExampleTest to use page not dependent on database.

SKroell committed:
Update deploy action

SKroell committed:
Fixed deploy action

Made a mistake... on the last one

SKroell committed:
Use ssh command instead of rsync

THHamiltonSmith committed:
Added .DS_Store to .gitignore

.DS_Store is a useless file that macOS automatically generates in all folders to save the positions of folders in the directory, it is useless on windows and is an annoying file.

PotatoPlank committed:
Dependency updates

PotatoPlank committed:
Database migrations & Cleanup
Users, Group, and OAuth/Passport table migrations.
Passport configuration.
Removed default User controller

PotatoPlank committed:
User and Group models.
Added models and API Routes for Passport

PotatoPlank committed:
Added Controllers
Added API Auth, Group, and Web Auth controllers.

PotatoPlank committed:
View changes
Removed default welcome route.
Added basic profile view and route.

SKroell committed:
On mobile devices 'Dictionary' menu label was incorrectly labeled as 'canon'

SKroell committed:
Fixed app.php using wrong environment values

PotatoPlank committed:
Added htaccess rules for word and dictionary pages
This commit fixes a problem presented by the Laravel conversion. Old search engine indexing and previous user bookmarks would present a 404 error. The word and dictionary pages will now properly redirect to the new routes.

PotatoPlank committed:
Fixed wrong variable referenced for G-app Analytics

PotatoPlank committed:
Added route redirect for URL's that match word?q=*

PotatoPlank committed:
Removed htaccess rules and enacted routes for old path redirects

PotatoPlank committed:
Fixed missing/broken references for app config indexes

ZoeS17 committed:
[FIX] Issue #25 trig->trig not trig->English

*FIX
Anchor en_text should now display English correctly again

ZoeS17 committed:
[FIX] Issue-13

*FIX
Adds noncanon-warning to dictionary view. Fixing issue #13

ZoeS17 committed:
[FIX] Limit scope of noncanon-warning fixing commit c9b219b

*FIX
Fix my oversight as per requested. Hopefully finally fixing this issue.

ZoeS17 committed:
Update resources/views/dictionary/lookup.blade.php

Co-authored-by: Stefan Kröll <stefan@stemaki.dk>

SKroell committed:
Fix #28

PotatoPlank committed:
Optimizations related to routes and moved functionality to API calls.

* Validation failures should now use the 422 HTTP error code. This signifies a structurally valid request, but bad data.
* All Authentication should now be contained in the Web\AuthController and uses the new API.
* Cleaned up the "repository" class WebController.php
* Updated main.js to redirect to the homepage on signup and to support password confirmation fields.
* Removed profile icon from navbar. This isn't finalized and should be hidden.

PotatoPlank committed:
Minor updates and Group List endpoint
* Updated existing API routes with names
* Added Groups list endpoint. (GET /groups)

PotatoPlank committed:
Add new routes and gates
* Added User routes (GET, PUT) for getting a user profile and updating.
* Added Gates to consolidate access validation.

SKroell commented on app/Http/Controllers/Api/v1/AuthController.php:
```diff
@@ -0,0 +1,113 @@
+     */
+    public function signup(Request $request) {
+        $validator = Validator::make($request->all(), [
+            'username' => 'required|string',
```
Username should be unique as well,

PotatoPlank replied:
Good point! I'll update that.

dependabot[bot] committed:
Bump websocket-extensions from 0.1.3 to 0.1.4

Bumps [websocket-extensions](https://github.com/faye/websocket-extensions-node) from 0.1.3 to 0.1.4.
- [Release notes](https://github.com/faye/websocket-extensions-node/releases)
- [Changelog](https://github.com/faye/websocket-extensions-node/blob/master/CHANGELOG.md)
- [Commits](https://github.com/faye/websocket-extensions-node/compare/0.1.3...0.1.4)

Signed-off-by: dependabot[bot] <support@github.com>

SKroell commented on database/migrations/2016_06_01_000001_create_oauth_auth_codes_table.php:
```diff
@@ -0,0 +1,62 @@
+<?php
```
Is there a reason migrations are dated 06/01/2016? Not that it changes anything

PotatoPlank replied:
It's just the default date provided by Laravel Passports. Not really sure why! Maybe it's the date it was launched?

SKroell reviewed (commented):
Looks good :)

ZoeS17 commented on app/Http/Controllers/Web/AuthController.php:
```diff
@@ -0,0 +1,159 @@
+               'username' => $body->username,
+               'admin' => $body->group->admin,
+            ]);
+            return 'Success';
```
Is this only returning a string because there is no profile landing yet or were was the intent to only log the user in silently.

PotatoPlank replied:
It's just because that's what it did in the legacy code. As the code gets cleaned up, the response will likely be updated to a JSON string.

ZoeS17 commented on app/Http/Controllers/Web/AuthController.php:
```diff
@@ -0,0 +1,159 @@
+            return 'Success';
+        }catch(Exception $ex){
+            //Guzzle throws an exception during certain HTTP status codes
+            return 'Incorrect password or username!';
```
Shouldn't this line be, `return 'Incorrect password or email!';` since the users' email is used to login?

PotatoPlank replied:
Correct! I'll make that adjustment, it was an oversight on my part when moving the code over to the new class.

SKroell committed:
Merge pull request #33 from ProjectArkadiaTeam/dependabot/npm_and_yarn/websocket-extensions-0.1.4

Bump websocket-extensions from 0.1.3 to 0.1.4

SKroell committed:
Merge branch 'develop'

ZoeS17 committed:
[FIX] Translation edit page breaks on insert

*FIX
Make speaker and episode silently fail should their fields be left blank.

ZoeS17 committed:
[Fix] Update dict_translations->episode to default to 'other'

*Fix
Update the episode field of dict_translations to default to 'other' instead of an empty string or NULL.

SKroell committed:
Merge pull request #37 from ZoeS17/hotfix/issue36

[FIX] Translation edit page breaks on update

SKroell committed:
React revamp initial commit

SKroell committed:
Added word lookup page and api function

SKroell committed:
remove jQuery dependency and load js after pageload

SKroell committed:
Fix iOS issue and text hugging left side of the screen

SKroell committed:
Fixed site crash when word source = null

PotatoPlank committed:
Created initial migrations for new database structure.

PotatoPlank committed:
Fixed random character in sidebar.

SKroell committed:
Filter translations by season

SKroell committed:
Only show etymology and leipzig if available

SKroell committed:
Cleanup & Live search

SKroell committed:
Fix live search (woops)

SKroell committed:
Live search for translations page

SKroell committed:
Add S6 audio and script to auto match audio

SKroell committed:
Filter dictionary by word class

SKroell committed:
Show multiple word etymologies on one page

SKroell committed:
Reimplement "Recently Added" and random entries on home page

SKroell committed:
React authentication and part merge of PR #32

SKroell committed:
Accidentally broke translations page

SKroell committed:
Models and migrations for testing

PotatoPlank committed:
Current code for merging/comparing

SKroell committed:
Merge branch 'feature/react-ui' into feature/users

SKroell committed:
Cleanup migrations
